torch/testing/_internal/common_methods_invocations.py

```diff
@@ -14542,6 +14542,19 @@ def ref_pairwise_distance(input1, input2):
                    supports_forward_ad=True,
                    supports_fwgrad_bwgrad=True,
                    safe_casts_outputs=True),
+    UnaryUfuncInfo('special.log_ndtr',
+                   aten_name='special_log_ndtr',
+                   ref=scipy.speical.log_ndtr if TEST_SCIPY else _NOTHING,
+
+                   # TODO(bahaung): check following values
+                   decorators=(precisionOverride({torch.bfloat16: 5e-3,
+                                                  torch.float16: 5e-4}),),
+                   dtypes=all_types_and(torch.bool, torch.bfloat16),
```

Comment:
I couldn't find where the supported input dtypes are specified? It looks like the ops in pytorch doesn't have a strong-typed schema?

Reply:
scipy's impl supports complex input type, do we also need to support complete type?

Reply:
The PR specifies the input datatypes using the dispatch macro and the wrapper. The dispatch macro defines the kernel for float and double, and then the wrapper specifies that boolean and integer dtypes should promote to a float type before the kernel is called. From what I'm reading I wouldn't think that bfloat16 or float16 are supported. Where are they coming from?

Reply:
indeed, these are not needed.